tests/gene-details.spec.ts

```diff
@@ -7,11 +8,12 @@ test.describe('specific viewport block', () => {
   test('invalid gene results in a 404 redirect', async ({ page }) => {
     // go to invalid ENSG page
     await page.goto('/genes/ENSG00000000000');
+    await waitForSpinnerNotVisible(page);
 
     // expect a title "to contain" a substring.
     await expect(page).toHaveTitle('Agora');
 
     // expect div for page not found content to be visible
-    expect(page.locator('.page-not-found')).toBeVisible();
+    await expect(page.locator('.page-not-found')).toBeVisible();
```

Comment:
Fix warning from `@typescript-eslint/no-floating-promises` -- toBeVisible is an [auto-retrying assertion](https://playwright.dev/docs/test-assertions#auto-retrying-assertions)